app/models/webhook.rb

```diff
@@ -81,19 +90,50 @@ def ca_certs_store
 
   private
 
+  def http_headers_json
+    return if http_headers.blank?
+
+    JSON.parse(http_headers)
+  rescue JSON::ParserError => e
+    errors.add(:http_headers, _('is not valid JSON %s') % e)
+  end
```

Comment:
Yeah you are right that this validation is not correct. I was testing it only with data looking like this:

```json
{
"test: "<%= 1+1 %>"
}
```

and that worked well.

I don't know I don't want to be adding N attributes for N http headers. I could render the field and then check the output but what if it contains some macro that is costly to execute? Not so sure about this.

Reply:
I have to remove the validator, I cannot render headers without proper context (event, payload). We need to raise an error during actual delivery.